crates/engine/tree/src/tree/root/sync_task.rs

```diff
@@ -0,0 +1,82 @@
+//! Sync state root task implementation.
+
+use super::{StateRootConfig, StateRootHandle, StateRootResult};
+use reth_trie::updates::TrieUpdates;
+use revm_primitives::{EvmState, B256};
+use std::sync::mpsc::{self, Receiver, RecvError};
```

Comment:
worth checking if we get perf improvements by using crossbeam channels

Reply:
ok, I've added a bench here https://github.com/paradigmxyz/reth/pull/12378/commits/e2dd3ad3ef4bec53c07f0c1bf149cf134a86b078 to compare `crossbeam_channel::bounded(1)` and `std::sync::mpsc::sync_channel(1)` when sending `EvmState` (just sending items, not taking into account how they are processed), these are the results:

```
state_root_channels/std_channel/1
                        time:   [1.7216 ms 1.7296 ms 1.7401 ms]
state_root_channels/crossbeam_channel/1
                        time:   [564.40 µs 615.11 µs 652.70 µs]
state_root_channels/std_channel/10
                        time:   [1.8865 ms 2.0034 ms 2.0640 ms]
Found 1 outliers among 10 measurements (10.00%)
  1 (10.00%) high mild
state_root_channels/crossbeam_channel/10
                        time:   [749.44 µs 803.68 µs 867.41 µs]
state_root_channels/std_channel/100
                        time:   [3.2193 ms 3.2717 ms 3.3067 ms]
state_root_channels/crossbeam_channel/100
                        time:   [2.9744 ms 3.0334 ms 3.1140 ms]
```
So, switching to crossbeam makes great improvement for small state changes (1-10 accounts changed, as we probably will have in most of the transactions) and smaller improvement for bigger state changes; it never performs worse.

So looks like switching to crossbeam will be beneficial, wdyt?

Reply:
sorry for the previous message, i was benching the wrong thing (the channel for sending the final result instead of the channel for sending each state update).

ok, we have the right bench here https://github.com/paradigmxyz/reth/pull/12378/commits/72395388915dc984ffab1ba4b3eccb311f6e0d6a to compare `crossbeam_channel::unbounded` and `std::sync::mpsc::channel()` when sending `EvmState` (just sending items, not taking into account how they are processed), these are the results:

```
state_stream_channels/std_channel/1
                        time:   [49.237 µs 50.471 µs 51.552 µs]
state_stream_channels/crossbeam_channel/1
                        time:   [47.803 µs 50.518 µs 55.016 µs]
state_stream_channels/std_channel/10
                        time:   [286.21 µs 293.42 µs 297.39 µs]
Found 4 outliers among 10 measurements (40.00%)
  1 (10.00%) low severe
  1 (10.00%) low mild
  2 (20.00%) high mild
state_stream_channels/crossbeam_channel/10
                        time:   [285.84 µs 295.41 µs 306.75 µs]
state_stream_channels/std_channel/100
                        time:   [1.0524 ms 1.0585 ms 1.0670 ms]
Found 1 outliers among 10 measurements (10.00%)
  1 (10.00%) high mild
state_stream_channels/crossbeam_channel/100
                        time:   [1.0317 ms 1.0411 ms 1.0510 ms]
```
so, mostly the same performance for both implementations, slightly better crossbeam for big state changes (100 accounts), but this is not going to be very frequent.

given these results I think it would be better to keep `std::syn::mpsc::channel`, one dep less, wdyt?

Reply:
I see, then this doesn't really matter.
crossbeam is already part of the dep tree so we can use it or not